src/common/utils/match-route-order-independent/utils.tsx

```diff
@@ -0,0 +1,69 @@
+import type { Match, Query } from '../../types.tsx';
+
+const hasOwnProperty = Object.prototype.hasOwnProperty;
+const MAX_CACHE_SIZE = 1000;
+
+function shallowEqual(objA: any, objB: any) {
```

Comment:
Is this a copied from somewhere? I remember seeing something similar already in the repo.

Reply:
Yes, it's a duplicate. I copied it because I want to keep the cache separate between the existing matchRoute and the order-independent matchRoute.
Thinking about it, I can create a factory function to avoid duplicated code. Let me do that.